infra/data-pipeline.yml

```diff
@@ -1,455 +1,494 @@
-           taskboot --cache /cache --target /code build-compose --registry=registry.hub.docker.com --write /images --service bugbug-http-service-bg-worker --tag ${version} --tag latest --build-arg BUGBUG_VERSION=${version}"
-      artifacts:
-        public/bugbug:
-          expires: {$fromNow: '2 weeks'}
-          path: /images
-          type: directory
-    scopes:
-      - docker-worker:capability:privileged
-
-    routes:
-      - notify.email.release-mgmt-analysis@mozilla.com.on-failed"
-      - notify.irc-channel.#bugbug.on-failed
-    metadata:
-      name: bugbug docker http build
-      description: bugbug docker http build
-      owner: release-mgmt-analysis@mozilla.com
-      source: https://github.com/mozilla/bugbug/raw/master/data-pipeline.yml
-
-  - ID: docker-push
-    dependencies:
-      - docker-build
-    scopes:
-      - secrets:get:project/relman/bugbug/deploy
-    created: {$fromNow: ''}
-    deadline: {$fromNow: '120 hours'}
-    expires: {$fromNow: '1 month'}
-    provisionerId: aws-provisioner-v1
-    workerType: github-worker
-    payload:
-      features:
-        taskclusterProxy:
-          true
-      maxRunTime: 3600
-      image: mozilla/taskboot:0.1.7
-      env:
-        TASKCLUSTER_SECRET: project/relman/bugbug/deploy
-      command:
-        - taskboot
-        - push-artifact
-
-    routes:
-      - notify.email.release-mgmt-analysis@mozilla.com.on-failed"
-      - notify.irc-channel.#bugbug.on-failed
-    metadata:
-      name: bugbug docker http push
-      description: bugbug docker http push
-      owner: release-mgmt-analysis@mozilla.com
-      source: https://github.com/mozilla/bugbug/raw/master/data-pipeline.yml
-
-  - ID: http-deploy
-    dependencies:
-      - docker-build
-    scopes:
-      - secrets:get:project/relman/bugbug/deploy
-    created: {$fromNow: ''}
-    deadline: {$fromNow: '120 hours'}
-    expires: {$fromNow: '1 month'}
-    provisionerId: aws-provisioner-v1
-    workerType: github-worker
-    payload:
-      features:
-        taskclusterProxy:
-          true
-      maxRunTime: 3600
-      image: mozilla/taskboot:0.1.7
-      env:
-        TASKCLUSTER_SECRET: project/relman/bugbug/deploy
-      command:
-        - "/bin/sh"
-        - "-lcxe"
-        - "taskboot deploy-heroku --artifact-filter public/bugbug/bugbug-http-service.tar --heroku-app bugbug &&
-           taskboot deploy-heroku --artifact-filter public/bugbug/bugbug-http-service-bg-worker.tar --heroku-app bugbug --heroku-dyno-type worker"
-
-    routes:
-      - notify.email.release-mgmt-analysis@mozilla.com.on-failed"
-      - notify.irc-channel.#bugbug.on-failed
-    metadata:
-      name: bugbug docker http service deploy
-      description: bugbug docker http service deploy
-      owner: release-mgmt-analysis@mozilla.com
-      source: https://github.com/mozilla/bugbug/raw/master/data-pipeline.yml
+  $let:
+    now: {$eval: 'now'}
```

Comment:
What if you define year, month, etc. too here?

Reply:
I'm not sure if I define them as:

```diff
   $let:
-    now: {$eval: 'now'}
+    year: {$eval: 'now[0:4]'}
+    month: {$eval: 'now[5:7]'}
```

that `now` will be the same datetime object. For year and month it shouldn't be a problem but for minute and seconds it might be problematic.

Reply:
Maybe you can do another let:
```
$let:
    now: {$eval: 'now'}
in:
  $let:
    year: $now[0:4]
...
```

Reply:
I checked locally and it seems fine, I hope json-e implementation used by the taskcluster instance behaves the same